ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/application/hooks/CommonHooks.kt

```diff
@@ -127,3 +127,20 @@ public object ReceiveRequestBytes : Hook<(call: ApplicationCall, body: ByteReadC
         }
     }
 }
+
+/**
+ * A hook that is invoked before `Transform` phase.
+ * Useful for some plugins which used for templates as views within application.
+ */
+public object ResponseBeforeTransform : Hook<suspend PipelineContext<Any, ApplicationCall>.(body: Any) -> Any> {
```

Comment:
a couple of things
1. One of the goals to design new plugins API was to hide all the `Pipeline` and `PipelineContext` related entities from the users. This hook exposes them.
2. I am not sure if it should be public even. I'm ok with duplicating it in every related plugin to keep in internal. If you decide to keep it public, we need a better name for it.

Reply:
okay, I think it's better not to expose `Pipeline` and `PipelineContext`, I'll rewrite it